### PR TITLE
CI: remove push contraints for tests

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

When commit gets pushed to the repo before it only ran the tests related to that commit. We had some issues doing this because of dependencies between packages in the same repo. To avoid changing stuff in one package breaking stuff in the other packages we test everything on all commits 

Task ID: [OS-1086](https://aragonassociation.atlassian.net/browse/OS-1086)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

[OS-1086]: https://aragonassociation.atlassian.net/browse/OS-1086?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sdk/test/unit/context.test.ts [hunk 1]
```diff
@@ -284,19 +284,12 @@ describe('Context instances', () => {
...
-          );
-        })
-        .catch(err => {
-          throw err;
-        });
+      const network = provider.network;
+      expect(network.chainId).toEqual(84531);
+      expect(network.name).toEqual('baseGoerli');
+      expect(network.ensAddress).toEqual(
+        LIVE_CONTRACTS[SupportedVersion.LATEST].baseGoerli.ensRegistryAddress
```
Comment (jordaniza) [inline sdk/test/unit/context.test.ts:291]: why is this changed in the PR?

Comment (josemarinas) [reply sdk/test/unit/context.test.ts:291]: Used to be handled with a then, for some reason that failed so i deleted it, then the indent on this changed that why its showing up as changes

Comment (Michael-A-Heuer) [reply sdk/test/unit/context.test.ts:291]: Is this the run that failed? https://github.com/aragon/osx-commons/actions/runs/8095266336/job/22121537332#step:6:176

I encountered the same error [in another PR](https://github.com/aragon/osx-commons/actions/runs/8065527304/job/22031515424#step:6:176) and mentioned it in our [discord thread here](https://discord.com/channels/672466989217873929/1212050307530367027/1212050309115940966).
I would like to understand this more. Is this really fixed with this change?

**Review by Michael-A-Heuer — CHANGES_REQUESTED** [review]
Let's figure out was is causing the SDK tests to fail.

File: sdk/test/unit/context.test.ts (hunk 1 above)
Comment (josemarinas) [reply sdk/test/unit/context.test.ts:291]: Yes, this fixes the issue, however this is part of the sdk 1.0. I plan to remove this from thee commons after I have all tested and sorted

**Review by jordaniza — APPROVED** [review]

**Review by Michael-A-Heuer — APPROVED** [review]
